src/azul/strings.py

```diff
@@ -232,11 +232,11 @@ def jl(*lines: str) -> str:
     return '\n'.join(lines)
 
 
-def jw(*words: str) -> str:
+def jw(*words: str | None) -> str:
```

Comment:
I don't think this is a good idea. Filtering and joining the strings in the same function was okay when it was a local function that could only be used in one place, but a public function like this shouldn't be so fine-tuned to a single call site. There may be other call sites someday where the filtering isn't necessary or appropriate (it could hide bugs, for example).

Reply:
If I understand correctly, the only word that actually needs to be filtered is the `nonce` variable. How about this approach instead to eliminate the call to `filter` altogether?

```diff
Subject: [PATCH] review
---
Index: src/azul/chalice.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/chalice.py b/src/azul/chalice.py
--- a/src/azul/chalice.py	(revision a36f7c5062b20fb1b79c21140c5bdb2c8db92fdf)
+++ b/src/azul/chalice.py	(date 1723684328983)
@@ -478,14 +478,13 @@
     def content_security_policy(self, nonce: str | None = None) -> str:
         self_ = sq('self')
         none = sq('none')
-        if nonce is not None:
-            nonce = sq('nonce-' + nonce)
+        nonce = [] if nonce is None else [sq('nonce-' + nonce)]
 
         return ';'.join([
             jw('default-src', self_),
             jw('img-src', self_, 'data:'),
-            jw('script-src', self_, nonce),
-            jw('style-src', self_, nonce),
+            jw('script-src', self_, *nonce),
+            jw('style-src', self_, *nonce),
             jw('frame-ancestors', none),
         ])
```